File: Frontend/src/services/supplierService.js
import api from "./api";

export const getAllSuppliers = () => {
  return api.get("/supplier/getall");
};

export const addSupplier = (supplier) => {
  return api.post("/supplier/add", supplier);
};

export const updateSupplier = (id, supplier) => {
  return api.put(`/supplier/update/${id}`, supplier);
};

export const deleteSupplier = (id) => {
  return api.delete(`/supplier/delete/${id}`);
};

File: Backend/medicalstoremanagement/src/main/java/com/medicalstore/service/SupplierService.java
package com.medicalstore.service;

import java.util.List;

import com.medicalstore.model.Supplier;

public interface SupplierService {

	Supplier addSupplier(Supplier supplier);

	List<Supplier> getAllSuppliers();

	Supplier getSupplierById(Long supplierId);

	Supplier updateSupplier(Long supplierId, Supplier supplier);

	void deleteSupplier(Long supplierId);

}